### PR TITLE
Use OTEL or ELASTIC_APM env variables

### PR DESCRIPTION
**Pull request by Mpdreamz**
Based on either OTEL environment variables or Elastic APM with OTEL having precedence.

This also ensures `CreateNewWithDefaults` allows to be cached busted by providing a `local` cache object.

Fixes: https://github.com/elastic/apm-agent-dotnet/issues/2033
Relates: https://github.com/elastic/apm/pull/813

### CODE REVIEW

**Review by stevejgordon — CHANGES_REQUESTED** [review]
1 question, 1 suggestion and 1 change.

File: src/Elastic.CommonSchema/EcsDocument.DefaultService.cs [hunk 1]
```diff
@@ -6,17 +8,52 @@ namespace Elastic.CommonSchema;
...
+			service.Name = resourceServiceName;
+		else if (!string.IsNullOrEmpty(apmServiceName))
+			service.Name = apmServiceName;
+
+		//ServiceVersion
+		var apmServiceVersion = Environment.GetEnvironmentVariable("ELASTIC_APM_SERVICE_VERSION");
+		if (resourceAttributes.TryGetValue("service.version", out var resourceServiceVersion))
+			service.Version = resourceServiceVersion;
+		else if (!string.IsNullOrEmpty(apmServiceName))
+			service.Version = apmServiceName;
```
[the remark this answers is not in the document]
Comment (Mpdreamz) [reply src/Elastic.CommonSchema/EcsDocument.DefaultService.cs:42]: Yes, facepalm on my end :)

File: src/Elastic.CommonSchema/EcsDocument.cs [hunk 2]
```diff
@@ -21,22 +23,45 @@ public interface IEcsDocumentCreationOptions
...
+
+	/// <summary> Cached parsed OpenTelemetry resource attributes </summary>
+	public IDictionary<string, string>? OTelResourceAttributes { get; set; }
 }
 
 public partial class EcsDocument
 {
-	private static readonly Ecs EcsFieldDefault = new() { Version = Version };
+	private static readonly EcsDocumentCreationCache DefaultCache = new()
+	{
```
[the remark this answers is not in the document]
Comment (Mpdreamz) [reply src/Elastic.CommonSchema/EcsDocument.cs:62]: Yeah the private set on first invocation is expected and what allows others to inject their own.

**Review by stevejgordon — APPROVED** [review]
LGTM

**Pull request merged**